araproc/framework/map_utilities.py

```diff
@@ -1,74 +1,296 @@
+        """
+        # Use numpy array and pass as three arguments
+        input_coords = np.array(array_coords, dtype=np.float64)
+        output_coords = np.zeros(3, dtype=np.float64)
+        geomTool.convertArrayToStationCoords(station_id, input_coords, output_coords)
+        return output_coords
+
+    def get_southpole_landmarks(landmark_type):
+        """
+        Currently We have included three known landmarks at southopole
+        Parameters
+        ----------
+        landmark_type : str
+           ICL : stands for IceCube Lab
+           WT : Wind Turbine
+           SPT : South Pole Telescope   
+        Returns
+        -------
+        landmarks : array
+            The landmark location in ARA Gloabal (Array) Coordinate system
+        """ 
+        landmarks = []
+        if landmark_type == "ICL":
+            for i in range(1): ## This can go range(4)
+                corner = geomTool.getICLCorner(i)
+                landmarks.append([corner[0], corner[1], corner[2]])
+        elif landmark_type == "WT": ## This can go range(1,4)
+            for i in range(3, 4):
+                corner = geomTool.getWindTurbine(i)
+                landmarks.append([corner[0], corner[1], corner[2]])
+        elif landmark_type == "SPT":
+            corner = geomTool.getSouthPoleTelescope()
+            landmarks.append([corner[0], corner[1], corner[2]])
+        else:
+            raise ValueError(f"Unknown landmark type: {landmark_type}")
+        return landmarks
+
+    def get_distant_pulsers(pulser_name,spice_depth = None):
+        """
+        returns the station centric coordinate of given distant pulser
+        Parameters
+        ----------
+        pulser_name: str
+          IC1S : IceCube 1S pulser
+          IC22S : IceCube 22S pulser
+          Spice : Spicecore pulser 
+        Returns
+        -------
+        st_centric_coords: array
+          X,Y,Z location of pulser in station centric
+        """
+        if pulser_name == "Spice" and spice_depth is None:
+           raise ValueError(f"Unknown Spice pulser depth")
+        pulser_coords = {
+            "IC1S": [45659.6457, 50490.4199],
+            "IC22S": [44884.416, 51444.8819],
+            "Spice": [42600, 48800],
+        }
+        pulser_depths = {"IC1S": -1400, "IC22S": -1450.47,"Spice": spice_depth}
+        if pulser_name not in pulser_coords:
+            raise ValueError(f"Unknown pulser name: {pulser_name}")
+
+        easting, northing = pulser_coords[pulser_name]
+        global_coords = get_survey_to_global_coords(easting, northing) * FEET_TO_METER
+        global_coords = [float(global_coords[0]), float(global_coords[1]), 0]
+        st_centric_coords = get_global_to_station_centric(global_coords)
+        return st_centric_coords[0], st_centric_coords[1], pulser_depths[pulser_name]
+
+    def get_cartesian_to_spherical(coord1, coord2): 
+        """
+        Parameters
+        ----------
+        coord1: array 
```

Comment:
What are coordinate 1 and 2 here? Seems like this is more than a conversion from cartesian to spherical? It seems like it also does a coordinate transformation?

Reply:
Lazy me! I am replacing coords 1 and 2 as reference_point and target_point. True! It's not simply a conversion. I can think of this function name : relative_cartesian_to_spherical? Do you have any suggestion?

Reply:
hmm maybe `get_relative_position` is enough? or `get_relative_spherical_position`